gammapy/data/ivoa.py

```diff
@@ -0,0 +1,327 @@
+    return tab_default
+
+
+def _obscore_row(base_dir, single_obsID, obs_publisher_did, access_url, table):
+    """Generates an obscore row corresponding to a single obsID
+
+    Parameters
+    ----------
+    single_obsID : int
+        single Observation ID
+    **kwargs : `str` {obs_publisher_did, access_url}
+    Giving the values for is highly recommended.
+    If any of these are not given the corresponding obscore field is left empty and a warning is raised for each empty value.
+
+    Returns
+    -------
+    table : `~astropy.table.Table`
+    """
+    base_dir = make_path(base_dir)
+    data_store = DataStore.from_dir(base_dir)
+
+    if obs_publisher_did is None:
+
+        log.warning(
+            "Insufficient publisher information: 'obs_publisher_did' obscore value will be empty."
+        )
+    if access_url is None:
+        log.warning(
+            "Insufficient publisher information: access_url' obscore value will be empty."
+        )
+
+    tab = table
+    observation = data_store.obs(single_obsID)
+    obs_mask = data_store.obs_table["OBS_ID"] == observation.obs_id
+    obs_pos = data_store.obs_table[obs_mask]["EVENTS_FILENAME"]
+    path = make_path(os.environ["GAMMAPY_DATA"] + "/" + obs_pos[0])
+    size = int(os.path.getsize(path) / 1000.0)
+    tab.add_row(
+        {
+            "dataproduct_type": observation.obs_info["EXTNAME"],
+            "calib_level": 2,  # Look into the data
+            "target_name": observation.obs_info["OBJECT"],
+            "obs_id": str(observation.obs_info["OBS_ID"]),
+            "obs_collection": "DL3",
+            "obs_publisher_did": str(obs_publisher_did),
+            "access_url": str(access_url),
+            "access_format": "application/fits",
+            "access_estsize": size,
+            "s_ra": observation.get_pointing_icrs(observation.tmid).ra.to_value("deg"),
+            "s_dec": observation.get_pointing_icrs(observation.tmid).dec.to_value(
+                "deg"
+            ),
+            "s_fov": 10.0,
+            "t_min": observation.tstart.to_value("mjd"),
+            "t_max": observation.tstop.to_value("mjd"),
+            "t_exptime": observation.observation_live_time_duration.to_value("s"),
+            "em_min": observation.events.energy.min().value,
+            "em_max": observation.events.energy.max().value,
+            "facility_name": observation.obs_info["TELESCOP"],
+            "instrument_name": observation.obs_info["TELLIST"],
+        }
+    )
+
+    return tab
+
+
+def to_obscore_table(
+    base_dir, selected_obs=None, obs_publisher_did=None, access_url=None
+):
+    """Generate the complete obscore Table by adding one row per observation using _obscore_row()
+
+    Parameters
+    ----------
+    selected_obs : list or array of Observation ID(int)
+    (default of ``None`` means ``no observation ``)
+    If not given, the obscore default table is returned.
+
+
+
+    **kwargs : `str` {obs_publisher_did, access_url}
```

Comment:
Maybe mention what `obs_publisher_did` is supposed to be. it is not intuitively obvious. Same for access_url